### PR TITLE
Add Aten::max_pool3d forward/backward

### PR DESCRIPTION
**Pull request by chunhuanMeng**
- [x] Aten:max_pool3d forward
- [x] Aten:max_pool3d backward

image

### CODE REVIEW

File: src/ATen/native/xpu/sycl/DilatedMaxPool3d.cpp [hunk 1]
```diff
@@ -0,0 +1,779 @@
...
+        channels_last);
+
+    // width size is fixed size = 32, height dim equals = syclMaxWorkGroupSize
+    // / width_size
+    int width_group_size = 32;
+    int height_group_size = syclMaxWorkGroupSize(kfn) / width_group_size;
+    int width_group_range = ceil_div<int>(owidth, width_group_size);
+    int height_group_range = ceil_div<int>(oheight, height_group_size);
+
+    int z_group_range = totalZ > 65535 ? 65535 : totalZ;
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/DilatedMaxPool3d.cpp:255]: 65535 is CUDA's restriction for grid size. SYCL doesn't need.
so we can 1, remove this loop.
2, use max work item per tile as global range per loop.

File: src/ATen/native/xpu/sycl/DilatedMaxPool3d.cpp [hunk 2]
```diff
@@ -0,0 +1,779 @@
...
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/DilatedMaxPool3d.cpp:584]: ditto

File: src/ATen/native/xpu/sycl/DilatedMaxPool3d.cpp [hunk 3]
```diff
@@ -0,0 +1,779 @@
...
+#include <comm/SYCLHelpers.h>
+
+namespace at::native::xpu {
+
+inline int min(int a, int b) {
+  return a <= b ? a : b;
+}
+
+template <typename scalar_t>
+struct MaxPool3dWithIndicesOutFrameImplKernelFunctor {
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/DilatedMaxPool3d.cpp:26]: MaxPool3dKerenlFunctor

File: src/ATen/native/xpu/sycl/DilatedMaxPool3d.cpp [hunk 4]
```diff
@@ -0,0 +1,779 @@
...
+            pW,
+            dilationT,
+            dilationH,
+            dilationW,
+            channels_last);
+      });
+}
+
+template <typename scalar_t>
+struct MaxPool3dWithIndicesBackwardOutFrameImplKernelFunctor {
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/DilatedMaxPool3d.cpp:449]: MaxPool3dBackwardKernelFunctor

**Review by xytintel — APPROVED** [review]

**Pull request merged**